File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpaceRuby/ShieldBooster.rb
require_relative 'ShieldToUI'

module Deepspace

    class ShieldBooster

        def initialize (name, boost, uses)
            @name = name
            @boost = boost
            @uses = uses
        end

        def self.newCopy (s)
            new(s.name, s.boost, s.uses)
        end

        def boost
            @boost
        end

        def uses
            @uses
        end

        def name
            @name
        end
        
        def useIt
            if (@uses > 0)
                @uses -= 1
                @boost
            else
                1.0
            end
        end

        def getUIversion
            return ShieldToUI.new(self)
        end 

        def to_s
            return getUIversion.to_s
        end
    end

end

File: 2/2_cuatrimestre/PDOO/practicas/practica 4/deepSpace_4/src/deepspace/ShieldBooster.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package deepspace;

/**
 *
 * @author manbolq
 */
public class ShieldBooster implements CombatElement{
    private String name;
    private float boost;
    private int uses;
    
    
    /**************************************************************************/
    /*************************    CONSTRUCTORES   *****************************/
    /**************************************************************************/
    
    ShieldBooster(String name, float boost, int uses){
        this.name = name;
        this.boost = boost;
        this.uses = uses;
    }
    
    ShieldBooster(ShieldBooster s){
        name = s.name;
        boost = s.boost;
        uses = s.uses;
    }
    
    
    /**************************************************************************/
    /**************************    CONSULTORES   ******************************/
    /**************************************************************************/
    
    float getBoost(){
        return boost;
    }
    
    @Override
    public int getUses(){
        return uses;
    }
    
    /**************************************************************************/
    /****************************    METODOS   ********************************/
    /**************************************************************************/
    
    @Override
    public float useIt(){
        if (uses > 0){
            uses--;
            return boost;
        }
        else
            return 1.0f;
    }
    
    
    @Override
    public String toString(){
        return getUIversion().toString();
    }
    
    
    ShieldToUI getUIversion(){
        return new ShieldToUI(this);
    }
}


